auth/src/main/java/org/aerogear/mobile/auth/metrics/AuthMetricsProvider.java

```diff
@@ -0,0 +1,21 @@
+package org.aerogear.mobile.auth.metrics;
+
+import android.content.Context;
+
+import org.aerogear.mobile.metrics.interfaces.MetricsProvider;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class AuthMetricsProvider extends MetricsProvider {
+    @Override
+    public String namespace() {
+        return "auth";
+    }
+
+    @Override
+    public JSONObject metrics(Context context) throws JSONException {
```

Comment:
@pb82 is JSONObject the best choice here? It leaves it very open to the structure of the object returned. Is there some structure that should be enforced?

Reply:
> Is there some structure that should be enforced?
That's just metrics data. JSONObject/HashMap will will work.
Structure will be enforced on metrics payload:
```
{
 "metricsName": { "JSONObject key values will be here" }
}
```

Reply:
good point. If we want to enforce a flat structure we could return `Map<String, String>` or `Map<String, ClassEnclosingAllowedTypes>`

Reply:
@pb82 Actually I want to revert my comment. JSONObject will be better as it will allow nested data.